src/utils/isCoordinatesOverlap.ts

```diff
@@ -0,0 +1,41 @@
+import type { Coordinate } from "types/coordinate";
+
+interface DOMRect {
+  top: number;
+  right: number;
+  bottom: number;
+  left: number;
+}
+
+const isElementsOverlap = (domRect1: DOMRect, domRect2: DOMRect) => {
+  return !(
+    domRect1.top > domRect2.bottom ||
+    domRect1.right < domRect2.left ||
+    domRect1.bottom < domRect2.top ||
+    domRect1.left > domRect2.right
+  );
+};
+
+const coordinateToDOMRect = (coordinate: Omit<Coordinate, "id">) => ({
+  top: coordinate.y,
+  left: coordinate.x,
+  right: coordinate.x + coordinate.width,
+  bottom: coordinate.y + coordinate.height,
+});
+
+const isCoordinatesOverlap = (
```

Comment:
Do you think there could be some naming improvements regarding `isElementsOverlap` and `isCoordinatesOverlap`?

Reply:
I'm not sure if the naming scope to focus on “selectionBlock” is better than now or not? like `isSelectionBlockOverlap`…
Do you have any suggestions?